src/common/system_signal_handling.c

```diff
@@ -180,7 +191,47 @@ void dt_set_signal_handlers()
     fprintf(stderr, "[dt_set_signal_handlers] error: signal(SIGSEGV) returned SIG_ERR: %i (%s)\n", errsv,
             strerror(errsv));
   }
-#endif
+#else
+  /*
+  Set up exception handler for backtrace on Windows
+  works when there is NO SIGSEGV handler installed
+
+  Must be set up _after_: InitializeMagick(),
+  as GraphicsMagick is overwriting SetUnhandledExceptionFilter
+  an all other signals in its init function
+  The first invokation of the dt_set_signal_handlers() is _before_ InitializeMagick(),
+  therefore no point to set up exception handler
+  It should be set up only at the second invokation of  dt_set_signal_handlers(),
+  as at that time InitializeMagick() is done, and exception handler will work
+  Eventually InitializeMagick() should be fixed upstream not to ignore existing exception handlers
+  */
+  if(_times_handlers_were_set > 1)
+  {
+    ExcHndlInit();
+
+    gchar *name_used;
+    int fout;
```

Comment:
put all this into new function, and just call it here. let's try to not make this shiny new file unnecessary messy :)